### PR TITLE
Add a dev container to mangata-e2e

### PR DESCRIPTION
**Pull request by EddyJLSH**
These files allow you to use a docker environment to manage system and project dependencies. Requirements to use this are `Docker`, `docker-compose`, and `ms-vscode-remote.remote-containers`. When you open the repo a pop-up will ask if you want to run within the container. Note: we need to manually build `cmake` to install `xoshiro` (a psudorandom number library for JS) which takes some time to build the first time but does cache. 

Further caveat, you need to have ran `docker-compose -f devops/dockerfiles/docker-compose.yml up` in advance to have the network set-up for the devcontainer to work too. This can be improved by using docker-compose to run both at the same time in the future.

### CODE REVIEW

**Comment by EddyJLSH** [comment]
@goncer Hey! This is ready to try out. The Valuation type fix has sneaked in by mistake which suppresses the warning in the test log. It's good to add but let me know if you'd rather do it through a separate PR :)

**Review by goncer — COMMENTED** [review]
Do you mind updating the Readme with some instructions?
Do you think that would be valuable having a docker image in the docker hub instead of building it locally?

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    && ./bootstrap \
+    && make \
+    && make install 
+
+# Global npm modules
+RUN su node -c "npm install -g jest jest-allure"
+
+# Export variables
+ENV TEST_PALLET_ADDRESS='5EYCAe5XGPRojsCSi9p1ZZQ5qgeJGFcTxPxrsFRzkASu6bT2'
+ENV TEST_SUDO_NAME='//Maciatko'
```
Comment (goncer) [inline .devcontainer/Dockerfile:23]: this can not be here. Plz, remove any Sudo reference

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ENV API_URL='ws://node_alice:9944'
+ENV TEST_USER_NAME='//Alice'
+ENV LOG_INFO='info'
+ENV UI_URL='https://staging.mangata.finance/'
+ENV MNEMONIC_POLK='dismiss addict reduce fitness install aisle creek they seek palace stereo trumpet'
```
Comment (goncer) [inline .devcontainer/Dockerfile:28]: Plz, remove the mnemonics too. The repo is public and I don't want anybody using this user for nasty things :)

**Review by goncer — APPROVED** [review]

**Comment by EddyJLSH** [comment]
Merge eddy/add-dev-container

**Pull request merged**